MapboxDirections/MBVisualInstructionComponent.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+
+
+/**
+ :nodoc:
+ Encompasses a single visual component for a `RouteStep`.
+ */
+@objc(MBVisualInstructionComponent)
+public class VisualInstructionComponent: NSObject, NSSecureCoding {
+    
+    /**
+     :nodoc:
+     Basical text representation of visual component of a `RouteStep`.
+     */
+    public let text: String
+    
+    
+    /**
+     :nodoc:
+     A more detailed representation of the a `RouteStep`. With components, it's possible to show an image representation of highway shields.
+     */
+    public let components: [VisualInstructionDetailComponent]
```

Comment:
Per https://github.com/mapbox/MapboxDirections.swift/pull/197#discussion_r148165905, I still don’t see a point to a component class that contains instances of another component class. Why this extra level of indirection, as opposed to putting `primaryText` and `primaryComponents` directly on VisualInstruction?